File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/3.go
package main
import "fmt"

func doisUns(s string) bool {
    estado := "q0"
    for _, char := range s {
        if estado == "q0" && char == '1' {
            estado = "q1"
        } else if estado == "q1" && char == '1' {
            estado = "q2"
        } else if estado == "q2" && char == '1' {
            estado = "q3"
        }
    }
    return estado == "q2"
}

func main() {
    fmt.Println(doisUns("1100"))  // True
    fmt.Println(doisUns("111"))   // False
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/3.java
public class AFDDoisUns {
    public static boolean doisUns(String s) {
        String estado = "q0";
        for (char c : s.toCharArray()) {
            if (estado.equals("q0") && c == '1') {
                estado = "q1";
            } else if (estado.equals("q1") && c == '1') {
                estado = "q2";
            } else if (estado.equals("q2") && c == '1') {
                estado = "q3";
            }
        }
        return estado.equals("q2");
    }

    public static void main(String[] args) {
        System.out.println(doisUns("1100"));  // True
        System.out.println(doisUns("111"));   // False
    }
}
